packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -286,24 +446,22 @@ export class BlobManager {
                 this.redirectTable.delete(blobId);
                 continue;
             }
-            this.blobIds.delete(blobId);
         }
     }
 
     public summarize(telemetryContext?: ITelemetryContext): ISummaryTreeWithStats {
-        // If we have a redirect table it means the container is about to transition to "Attaching" state, so we need
-        // to return an actual snapshot containing all the real storage IDs we know about.
-        const attachingOrAttached = !!this.redirectTable || this.runtime.attachState !== AttachState.Detached;
-        const blobIds = attachingOrAttached ? this.blobIds : this.detachedBlobIds;
+        // if storageIds is empty, it means we are detached and have only local IDs, or that there are no blobs attached
+        const blobIds = this.storageIds.size > 0 ? Array.from(this.storageIds) : Array.from(this.redirectTable.keys());
         const builder = new SummaryTreeBuilder();
         blobIds.forEach((blobId) => {
             builder.addAttachment(blobId);
         });
 
-        if (this.redirectTable && this.redirectTable.size > 0) {
+        if (this.redirectTable.size > blobIds.length) {
             builder.addBlob(
                 BlobManager.redirectTableBlobName,
-                JSON.stringify(Array.from(this.redirectTable.entries())),
+                // filter out identity entries
+                JSON.stringify(Array.from(this.redirectTable.entries()).filter(([k, v]) => k !== v)),
```

Comment:
Nit: Comment explains what is happeniding, but consider 'localId' and 'storageId' rather than 'k' and 'v' for extra clarity